lightly/api/download.py

```diff
@@ -38,84 +38,311 @@ def download_image(url: str, session: requests.Session = None) -> PIL.Image.Imag
         The downloaded image.
 
     """
+    def load_image(url, req):
+        with req.get(url=url, stream=True) as response:
+            image = PIL.Image.open(response.raw)
+            image.load()
+        return image
+
     req = requests if session is None else session
-    response = utils.retry(req.get, url=url, stream=True)
-    return PIL.Image.open(response.raw)
+    image = utils.retry(load_image, url, req)
+    return image
 
+if not isinstance(av, ModuleNotFoundError):
```

Comment:
This is not the ideal way of handling this problem because now if a user doesn't have `av` installed but wants to use these functions the error message is very confusing. If you look at line `24` we have a function `_check_av_available` which raises a well-formatted and informative error if `av` is not available. Also, the error you've experienced comes from typehints only so my suggested fix is:
- Change all typehints which use `av.XYZ` to their string counterpart (i.e. `"av.XYZ"`)
- Add a call to `_check_av_available()` to each function using `av`.